lib/tasks/migrate_trn_to_profiles.rake

```diff
@@ -1,18 +1,26 @@
 namespace :jobseeker_profile do
-  desc "Migrate Teacher Reference Number (TRN) from job_applications to jobseeker_profiles"
+  desc "Migrate Teacher Reference Number (TRN) from job_applications to jobseeker_profiles, creating profiles if necessary"
   task migrate_teacher_reference_number_to_profiles: :environment do
     batch_size = 500
 
-    JobseekerProfile.find_in_batches(batch_size: batch_size) do |profiles|
-      profiles.each do |profile|
-        latest_job_application = JobApplication
-                                   .where(jobseeker_id: profile.jobseeker_id)
-                                   .order(created_at: :desc)
-                                   .first
+    JobApplication
+      .select("DISTINCT job_applications.jobseeker_id, job_applications.teacher_reference_number_ciphertext")
+      .where.not(teacher_reference_number_ciphertext: [nil, ""])
+      .joins("LEFT JOIN jobseeker_profiles ON jobseeker_profiles.jobseeker_id = job_applications.jobseeker_id")
+      .where("jobseeker_profiles.teacher_reference_number_ciphertext IS NULL OR jobseeker_profiles.teacher_reference_number_ciphertext = ''")
+      .find_in_batches(batch_size: batch_size) do |applications|
+      applications.each do |application|
+        profile = JobseekerProfile.find_or_initialize_by(jobseeker_id: application.jobseeker_id)
 
-        if latest_job_application&.teacher_reference_number.present?
-          profile.update(teacher_reference_number: latest_job_application.teacher_reference_number)
-          puts "Migrated TRN for profile ID: #{profile.id}"
+        puts "Creating profile for jobseeker ID: #{application.jobseeker_id}" if profile.new_record?
+
+        profile.teacher_reference_number_ciphertext = application.teacher_reference_number_ciphertext
```

Comment:
This line makes me a little nervous - should we not just be using teacher_reference_number?